SmallRye Config property mapping mismatches from the property name in the generated config documentation

### Describe the bug

When having a `@ConfigMapping` as the following:

```java
@ConfigMap@ConfigMappingping(prefix = "app")
@ConfigRoot(phase = ConfigPhase.RUN_TIME)
public interface OptionalConfig {

    /**
     * Whether the connection manager should use same RM override
     */
    Optional<Boolean> isSameRMOverride();
}
```

And the following entry in the application.properties: 

```properties
app.is-same-rm-override=true
```

The application fails to startup with:

```java
Caused by: io.smallrye.config.ConfigValidationException: Configuration validation failed:
	SRCFG00050: app.is-same-rm-override in PropertiesConfigSource[source=file:/private/tmp/config-optional/target/classes/application.properties] does not map to any root
	at io.smallrye.config.ConfigMappingProvider.mapConfigurationInternal(ConfigMappingProvider.java:1004)
```

### Expected behavior

The entry is read and injected in the config mapping method

### Actual behavior

```
Caused by: io.smallrye.config.ConfigValidationException: Configuration validation failed:
	SRCFG00050: app.is-same-rm-override in PropertiesConfigSource[source=file:/private/tmp/config-optional/target/classes/application.properties] does not map to any root
	at io.smallrye.config.ConfigMappingProvider.mapConfigurationInternal(ConfigMappingProvider.java:1004)
```

### How to Reproduce?

1. Unzip [config-optional-main.zip](https://github.com/quarkusio/quarkus/files/13267180/config-optional-main.zip)

2. Run `mvn test`

### Output of `uname -a` or `ver`

macOS m1

### Output of `java -version`

JDK 17

### Quarkus version or git rev

3.5.0

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Apache Maven 3.9.2

### Additional information

_No response_

--- a/bom/application/pom.xml
+++ b/bom/application/pom.xml
@@ -51,7 +51,7 @@
         <microprofile-lra.version>2.0</microprofile-lra.version>
         <microprofile-openapi.version>3.1.1</microprofile-openapi.version>
         <smallrye-common.version>2.2.0</smallrye-common.version>
-        <smallrye-config.version>3.4.1</smallrye-config.version>
+        <smallrye-config.version>3.4.4</smallrye-config.version>
         <smallrye-health.version>4.0.4</smallrye-health.version>
         <smallrye-metrics.version>4.0.0</smallrye-metrics.version>
         <smallrye-open-api.version>3.7.0</smallrye-open-api.version>

